File: src/main/java/org/jetbrains/fortran/lang/psi/impl/FortranLabelDoStmtImpl.java
package org.jetbrains.fortran.lang.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.jetbrains.fortran.lang.psi.*;

import static org.jetbrains.fortran.lang.FortranTypes.COLON;
import static org.jetbrains.fortran.lang.FortranTypes.EOL;

public class FortranLabelDoStmtImpl extends FortranStmtImpl implements FortranLabelDoStmt {

    public FortranLabelDoStmtImpl(ASTNode node) {
        super(node);
    }

    @Override
    @NotNull
    public FortranLabel getLabel() { return findNotNullChildByClass(FortranLabel.class); }

    @Override
    @Nullable
    public FortranLabelDecl getLabelDecl() { return findNotNullChildByClass(FortranLabelDecl.class); }

    @Override
    @Nullable
    public FortranConstructName getConstructName() {
        return findChildByClass(FortranConstructName.class);
    }
    @Override
    @Nullable
    public FortranExpr getExpr() {
        return findChildByClass(FortranExpr.class);
    }

    @Override
    @Nullable
    public PsiElement getColon() {
        return findChildByType(COLON);
    }


    @Override
    @Nullable
    public PsiElement getEol() {
        return findChildByType(EOL);
    }

}

File: core/src/main/kotlin/org/jetbrains/fortran/lang/psi/impl/FortranLabelDoStmtImpl.kt
package org.jetbrains.fortran.lang.psi.impl

import com.intellij.lang.ASTNode
import com.intellij.psi.PsiElement
import org.jetbrains.fortran.lang.FortranTypes
import org.jetbrains.fortran.lang.psi.*

class FortranLabelDoStmtImpl(node: ASTNode?) : FortranStmtImpl(node), FortranLabelDoStmt {
    override val label: FortranLabel
        get() = findNotNullChildByClass(FortranLabel::class.java)
    override val labelDecl: FortranLabelDecl?
        get() = findNotNullChildByClass(FortranLabelDecl::class.java)
    override val constructNameDecl: FortranConstructNameDecl?
        get() = findChildByClass(FortranConstructNameDecl::class.java)
    override val expr: FortranExpr?
        get() = findChildByClass(FortranExpr::class.java)
    override val colon: PsiElement?
        get() = findChildByType(FortranTypes.COLON)
    override val eol: PsiElement?
        get() = findChildByType(FortranTypes.EOL)
}